### ISSUE TITLE
On the ratio of no-resign games

### ISSUE THREAD

**Issue by isty2e** [gcp/leela-zero#1781]
Currently resignation is disabled in 20% of self-play games, in contrary to 10% for AGZ. Unfortunately the average game length of games with resignation enabled and disabled are unknown, but let us make a naive assumption about the average lengths of games: 

* 200 for regular games 
* 600 for no-resign games. 

Then there are 200 pre-resignation moves and 80 post-resignation moves per game, so about 29% of the training data is filled with some semi-random moves played after the resignation threshold is reached. This means that the loss function is minimized to fit these data, which might be a bit wasteful.

Unless there is a concrete reason to keep resignation disabled for the 20% of the games, wouldn't it be considered to lower the ratio to 10% as in AGZ (with resignation threshold changed if necessary)? This will decrease the proportion of post-resignation moves to 16%, and the self-play game generation will be faster by 9.1%, under the assumptions made about the average game lengths above.

**Comment by l1t1** [comment]
my pc is slow, so I often stop a resign=0 game and start a new autogtp

**Comment by gjm11** [comment]
An alternative solution to the same problem is the one discussed in issue #1766: play the same number of no-resign games, but play them _faster_ once the point at which one player would resign has been reached. I think "post-resignation" moves could be made a lot more than 2x faster, and given how bad the moves are at present I don't think this would actually make them appreciably worse.

**Comment by gjm11** [comment]
Ah, I see. My apologies for misunderstanding, then. (My guess is that the decision to make 20% of games no-resign rather than some other fraction was taken deliberately -- but of course it's possible that GCP hadn't anticipated just how much longer those games would be and therefore what a large fraction of moves/positions they would make up.)

**Comment by maxinjapan** [comment]
I think the no-resign option should be applied to games such as [the most recent match win by best network b1c37640 vs eb59a0fa](http://zero.sjeng.org/viewmatch/b0fb6fe095f38aed1beeeefe2681de0a1fd90effc003f5cccaabd56393025999?viewer=wgo), in which W resigned despite it being a very close game. 
I understand (kind of) that the decision to resign or not depends on the probability of winning a game, and this probability can be extremely low even in very close games, but I cannot help wondering whether it would not be better to play until the end close games, rather than resigning them. Just a thought...

**Comment by Mardak** [comment]
Lowered to 10% https://github.com/gcp/leela-zero-server/commit/d96e0eedd75ad9e87ce3efac7c22bd80c259f609

**Comment by Mardak** [comment]
In terms of history, the initial github version of server.js had no-resign at 20%. There was some discussion about setting it to 25% back in November https://github.com/gcp/leela-zero/issues/135#issuecomment-347037478

That was probably copied from the autogtp 20% https://github.com/gcp/leela-zero/pull/231/files#diff-ddf82fb0482f87e3cef577577efaef85R42 (ignore the "resignpct=10 in 90%" in the PR title)

**Comment by Friday9i** [comment]
Excellent ; -).
As no resign games represented somewhere around 50% of the time used for selfplay (as I cross-checked on my computer), dividing the % by 2 from 20% to 10% should increase overall speed of generating games by 20% to 25%!

Note: the 20% of no-resign games currently takes about as much time as the 80% of standard games, hence they are on average 4 times slower. Hence, if standard games take 1 unit of time, no-r games takes 4 units. Before, to generate 100 games, we needed 80 units from 80 std games and 20x4=80 units from no-r games, ie a total of 160 units. Now it will be 90 + 10x4 = 130 units. So 160/130 = 1.23 more efficiency!
